user_config.c

```diff
@@ -0,0 +1,73 @@
+#include "user_config.h"
+#include <stdio.h>
+
+static uint8_t set_debug_level = 0, change_function = 0, temp_num = 0;
+static bool set_sensor_messages = false;
+static char save_type;
+
+// function takes in a message and parses it to see if max_debug_level() or allow_sensor_messages() needs to be changed
+void parse_usb_string(const char *input) {
+	uint8_t iterate = 0;
+	while (input[iterate] != '\0') {
+		switch (change_function) {
+			// Case 2: last check for if there has been a valid message sent
+			// --> if parse_usb_string has received a semicolon
+			case 2:
+				change_function = 0;
+				if (input[iterate] == ';') {
+					if (save_type == 'G')
+						set_debug_level = temp_num;
+					else if (save_type == 'S')
+						set_sensor_messages = temp_num;
+					else if (save_type == 'L') {
+						printf("Current Config: \n");
+						printf("\tMax debug level: %d\n", max_debug_level());
+						if (allow_sensor_messages())
+							printf("\tSensor messages are allowed!\n");
+						else
+							printf("\tSensor messages are not allowed!\n");
+					}
+					break;
+				} // @suppress("No break at end of case")
```

Comment:
Why is there no break here? I'm assuming it's intentional, but why?

Reply:
There is no break because if the current character being checked is a new message (not apart of the old message being checked), then the function still needs to check it in other parts of the function (i.e. if "S0G1;" is sent into the function, when 'G' is being checked, the function should recognize that "S0G" is not a valid message but should also recognize that 'G' could also be the start of another message). 
I also realized that this check doesn't happen if it is at the second level check (check for the number of the level), so I will change the function accordingly.